### PR TITLE
Mark waSCC and WASI Kubelets as NoSchedule

### PR DESCRIPTION
**Pull request by kflansburg**
I believe this is the correct taint, as `NoExecute` applies to Pods that are already running. Probably doesn't hurt to keep `NoExecute` here. Sadly this does  not resolve the `kube-proxy` issue as they seem to apply a wildcard `NoSchedule` toleration.

### CODE REVIEW

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]
I think this is the right thing to do. Can you also update all of the docs and demo manifests to reflect this new change?

**Comment by thomastaylor312** [comment]
Oh I think you’ll need to update the e2e tests as well. 

Marking this as breaking so we remember to put something in the release notes

File: tests/integration_tests.rs [hunk 1]
```diff
@@ -81,7 +81,7 @@ async fn verify_wascc_node(node: Node) -> () {
         .expect("node had no taints");
     let taint = taints
         .iter()
-        .find(|t| t.key == "kubernetes.io/arch")
+        .find(|t| (t.key == "kubernetes.io/arch") & (t.effect == "NoExecute"))
```
Comment (thomastaylor312) [inline tests/integration_tests.rs:84]: Any reason you are using bitwise and here over a boolean and?

**Review by thomastaylor312 — APPROVED** [review]
This is looking good now. I'd like @bacongobbler or @itowlson to review this before merge as well since it is going to break current users when they upgrade. Want to make sure we are ok with this

**Review by bacongobbler — APPROVED** [review]
LGTM! Can you clarify what is the breaking change here though? Looks to me like we're just adding another taint to the nodes. Does that break existing pods if they upgraded Krustlet to 0.6? Trying to determine the severity/impact of the breaking change here. Thanks!

**Comment by kflansburg** [comment]
Yep, as seen with our test cases after my first commit, any pod spec without the NoSchedule toleration will no longer schedule after this change. We should at least note in the release notes that people update their tolerations.

**Pull request merged**